x/vesting/keeper/integration_test.go

```diff
@@ -204,6 +204,235 @@ var _ = Describe("Clawback Vesting Accounts", Ordered, func() {
+	vestingAmtTotal := sdk.NewCoins(sdk.NewCoin(stakeDenom, vestingTotal))
+
+	// 6 month cliff
+	cliff := int64(6)
+	cliffLength := vestingLength * cliff
+	cliffAmt := sdk.NewCoins(sdk.NewCoin(stakeDenom, amt.Mul(sdk.NewInt(cliff))))
+	cliffPeriod := sdkvesting.Period{Length: cliffLength, Amount: cliffAmt}
+
+	// 12 month lockup
+	lockup := int64(12) // 12 year
+	lockupLength := vestingLength * lockup
+	lockupPeriod := sdkvesting.Period{Length: lockupLength, Amount: vestingAmtTotal}
+	lockupPeriods := sdkvesting.Periods{lockupPeriod}
+
+	// Create vesting periods with initial cliff
+	vestingPeriods := sdkvesting.Periods{cliffPeriod}
+	for p := int64(1); p <= periodsTotal-cliff; p++ {
+		vestingPeriods = append(vestingPeriods, vestingPeriod)
+	}
+
+	var (
+		clawbackAccount *types.ClawbackVestingAccount
+		vesting         sdk.Coins
+		vested          sdk.Coins
+		unlocked        sdk.Coins
+		free            sdk.Coins
+	)
+	grantee := sdk.AccAddress(tests.GenerateAddress().Bytes())
+	funder := sdk.AccAddress(tests.GenerateAddress().Bytes())
+	dest := sdk.AccAddress(tests.GenerateAddress().Bytes())
+
+	BeforeEach(func() {
+		s.SetupTest()
+		ctx := sdk.WrapSDKContext(s.ctx)
+
+		// Create and fund periodic vesting account
+		vestingStart := s.ctx.BlockTime()
+		testutil.FundAccount(s.app.BankKeeper, s.ctx, funder, vestingAmtTotal)
+
+		balanceFunder := s.app.BankKeeper.GetBalance(s.ctx, funder, stakeDenom)
+		balanceGrantee := s.app.BankKeeper.GetBalance(s.ctx, grantee, stakeDenom)
+		balanceDest := s.app.BankKeeper.GetBalance(s.ctx, dest, stakeDenom)
+		s.Require().True(balanceFunder.IsGTE(vestingAmtTotal[0]))
+		s.Require().Equal(balanceGrantee, sdk.NewInt64Coin(stakeDenom, 0))
+		s.Require().Equal(balanceDest, sdk.NewInt64Coin(stakeDenom, 0))
+
+		msg := types.NewMsgCreateClawbackVestingAccount(funder, grantee, vestingStart, lockupPeriods, vestingPeriods, true)
+
+		_, err := s.app.VestingKeeper.CreateClawbackVestingAccount(ctx, msg)
+		s.Require().NoError(err)
+
+		acc := s.app.AccountKeeper.GetAccount(s.ctx, grantee)
+		clawbackAccount, _ = acc.(*types.ClawbackVestingAccount)
+
+		// Check if all tokens are unvested and locked at vestingStart
+		vesting = clawbackAccount.GetVestingCoins(s.ctx.BlockTime())
+		vested = clawbackAccount.GetVestedOnly(s.ctx.BlockTime())
+		unlocked = clawbackAccount.GetUnlockedOnly(s.ctx.BlockTime())
+		s.Require().Equal(vestingAmtTotal, vesting)
+		s.Require().True(vested.IsZero())
+		s.Require().True(unlocked.IsZero())
+
+		bF := s.app.BankKeeper.GetBalance(s.ctx, funder, stakeDenom)
+		balanceGrantee = s.app.BankKeeper.GetBalance(s.ctx, grantee, stakeDenom)
+		balanceDest = s.app.BankKeeper.GetBalance(s.ctx, dest, stakeDenom)
+
+		s.Require().True(bF.IsGTE(balanceFunder.Sub(vestingAmtTotal[0])))
+		s.Require().True(balanceGrantee.IsGTE(vestingAmtTotal[0]))
+		s.Require().Equal(balanceDest, sdk.NewInt64Coin(stakeDenom, 0))
+	})
+
+	It("clawback before cliff", func() {
```

Comment:
nit, the string message should more descriptive (in the form of `"should (not) clawback tokens ....")